content/usgs-packages/Application_Part3.Rmd

```diff
@@ -0,0 +1,182 @@
+                          colClasses = "character", stringsAsFactors = FALSE)
+sites <- sb_sites_df$site_number
+
+# get HUC 8 codes for precip data
+sb_sites_info <- readNWISsite(sites)
+huc8s <- sb_sites_info$huc_cd
+
+# define period
+startDate <- "2015-10-01"
+endDate <- "2016-09-30"
+
+# download nutrient data
+pcodes_nitrogen <- c("00613", "00618", "00631")
+pcodes_phosphorus <- c("00665")
+nitrogen_data <- readNWISqw(siteNumbers = sites, parameterCd = pcodes_nitrogen,
+                            startDate = startDate, endDate = endDate)
+phosphorus_data <- readNWISqw(siteNumbers = sites, parameterCd = pcodes_phosphorus,
+                              startDate = startDate, endDate = endDate)
+
+# download precip data
+precip_stencil <- webgeom(paste0('HUC8::', paste(huc8s, collapse=",")))
+precip_knife <- webprocess() # accept defaults for weighted average
+all_webdata <- query("webdata")
+precip_fabric <- webdata(all_webdata["United States Stage IV Quantitative Precipitation Archive"])
+variables(precip_fabric) <- query(precip_fabric, 'variables')
+times(precip_fabric) <- c(startDate, endDate)
+```
+
+Executing the geojob:
+
+```{r execute-job-off, addToggle="ON", eval=FALSE}
+precip_geojob <- geoknife(precip_stencil, precip_fabric, precip_knife)
+wait(precip_geojob, sleep.time = 10) # add `wait` when running scripts
+precip_data <- result(precip_geojob)
+```
+
+```{r read-intermediate, echo=FALSE}
+geoknife_list <- readRDS("precip_geoknife.RDS")
+precip_geojob <- geoknife_list$geojob
+precip_data <- geoknife_list$data
+```
+
+## Prepare data for plotting
+
+First re-organize the precipitation into a long format (one column for value, one for HUC) rather than a separate column of precip for each HUC. See `gather` from the package `tidyr` for ideas on how to do it.
+
+```{r reorganize-precip-data, addToggle="ON"}
+library(tidyr)
+precip_data_long <- gather(precip_data, huc8, precip, -which(!names(precip_data) %in% huc8s))
+```
+
+##
+Now choose one site to create the plots. Seperate the data so each data frame contains only one site (see `?filter` from `dplyr` for hints). After we establish the plotting code, we will automate this for all sites. 
+
+```{r filter-data, addToggle="ON", message=FALSE, warning=FALSE}
+library(dplyr)
+
+nitrogen_site1 <- filter(nitrogen_data, site_no == sites[1])
+phosphorus_site1 <- filter(phosphorus_data, site_no == sites[1])
+
+huc_site1 <- filter(sb_sites_info, site_no == sites[1])$huc_cd # corresponding HUC8
+precip_site1 <- filter(precip_data_long, huc8 == huc_site1)
+```
+
+## Create time series plots
+
+Create a separate time series plot of precipitation, nitrogen, and phosphorus for each site using the separated data frames. See `?layout` to see how to include all three in one graphics device. 
+
+```{r time-series-plots, addToggle="ON"}
+layout(matrix(1:3, nrow=3))
+plot(precip_site1$DateTime, precip_site1$precip,
+     col="red", pch=20, xlab = "Time", ylab = "Precip accumulation, in")
```

Comment:
Could this just be "Precip, in"? I'm not sure what "accumulation" means here, unless the data only reflect accumulated snow/surface water. From the plots it doesn't look like annual cumulative totals or anything.